vscode-extension/src/utils.ts

```diff
@@ -89,3 +89,50 @@ export async function findRelayBinary(
 
   return null;
 }
+
+export async function findRelayCompilerVersion(
```

Comment:
What do you think about having one function which returns both the binary and the version? Right now we have these theoretical issues where we have to consider what happens if we find the compiler binary but not the version, or the version but not the binary, when in reality we should always either find both or neither. Having a single function would ensure that:

1. We always get the version of the binary that we found, and we never somehow mess up and get the version from one place and the binary from another.
2. We can reason locally about how we want to handle the potential edge cases of finding one but not the other.

I think we should

1. traverse until we find `node_modules/relay-compiler` 
2. Check the version
3. If the version is modern enough, return that binary path

So maybe the unified function would return:

```
{kind: "compilerFound", path: string}
| {kind: "versionDidNotMatch", path: string, version: string} 
| {kind: "compilerNotFound" }
```